src/collections/btree.rs

```diff
@@ -0,0 +1,49 @@
+use crate::set::Set;
+use crate::update::Update;
+use std::collections::BTreeSet;
+use std::iter::FromIterator;
+
+pub struct BTreeSetStore<T: Ord + Clone> {
```

Comment:
Rather than writing type constraints within the <>s could we use the `where` style instead?